src/modules/ipfix/aggregator/BaseHashtable.cpp

```diff
@@ -122,7 +122,19 @@ void BaseHashtable::createDataTemplate(Rule* rule)
 			fi->type = rf->type;
 			fi->offset = fieldLength;
 			fi->privDataOffset = 0;
-			fieldLength += fi->type.length;
+			fi->isVariableLength = (fi->type.length == 65535);
+
+			if (!fi->isVariableLength) {
+				fieldLength += fi->type.length;
+			}
+			// Variable length fields: Extract real length information
```

Comment:
This does not take account of any variable length encoded field other than IPFIX_TYPEID_basicList.
This will break all other fields that vermont encodes as variable length, and importantly vermont always encodes all fields of type IPFIX_BYTES_octetArray and IPFIX_BYTES_string as variable length.

This appears to include a lot of import fields:

```
egrep -r "IPFIX_BYTES_octetArray|IPFIX_BYTES_string" ipfix_iana.h
#define IPFIX_BYTES_octetArray                65535
#define IPFIX_BYTES_string                    65535
#define IPFIX_LENGTH_mplsTopLabelStackSection                 IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection2                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection3                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection4                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection5                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection6                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection7                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection8                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection9                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection10                  IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_interfaceName                            IPFIX_BYTES_string
#define IPFIX_LENGTH_interfaceDescription                     IPFIX_BYTES_string
#define IPFIX_LENGTH_mplsVpnRouteDistinguisher                IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_applicationDescription                   IPFIX_BYTES_string
#define IPFIX_LENGTH_applicationId                            IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_applicationName                          IPFIX_BYTES_string
#define IPFIX_LENGTH_wlanSSID                                 IPFIX_BYTES_string
#define IPFIX_LENGTH_paddingOctets                            IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_VRFname                                  IPFIX_BYTES_string
#define IPFIX_LENGTH_metroEvcId                               IPFIX_BYTES_string
#define IPFIX_LENGTH_messageMD5Checksum                       IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_opaqueOctets                             IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_collectorCertificate                     IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_exporterCertificate                      IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_natPoolName                              IPFIX_BYTES_string
#define IPFIX_LENGTH_p2pTechnology                            IPFIX_BYTES_string
#define IPFIX_LENGTH_tunnelTechnology                         IPFIX_BYTES_string
#define IPFIX_LENGTH_encryptedTechnology                      IPFIX_BYTES_string
#define IPFIX_LENGTH_observationDomainName                    IPFIX_BYTES_string
#define IPFIX_LENGTH_ipHeaderPacketSection                    IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_ipPayloadPacketSection                   IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_dataLinkFrameSection                     IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsLabelStackSection                    IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mplsPayloadPacketSection                 IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_selectorName                             IPFIX_BYTES_string
#define IPFIX_LENGTH_informationElementDescription            IPFIX_BYTES_string
#define IPFIX_LENGTH_informationElementName                   IPFIX_BYTES_string
#define IPFIX_LENGTH_virtualStationInterfaceId                IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_virtualStationInterfaceName              IPFIX_BYTES_string
#define IPFIX_LENGTH_virtualStationUUID                       IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_virtualStationName                       IPFIX_BYTES_string
#define IPFIX_LENGTH_userName                                 IPFIX_BYTES_string
#define IPFIX_LENGTH_applicationCategoryName                  IPFIX_BYTES_string
#define IPFIX_LENGTH_applicationSubCategoryName               IPFIX_BYTES_string
#define IPFIX_LENGTH_applicationGroupName                     IPFIX_BYTES_string
#define IPFIX_LENGTH_dot1qServiceInstanceTag                  IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mibObjectValueOctetString                IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mibObjectValueOID                        IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mibObjectValueBits                       IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mibObjectIdentifier                      IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mibContextEngineID                       IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_mibContextName                           IPFIX_BYTES_string
#define IPFIX_LENGTH_mibObjectName                            IPFIX_BYTES_string
#define IPFIX_LENGTH_mibObjectDescription                     IPFIX_BYTES_string
#define IPFIX_LENGTH_mibObjectSyntax                          IPFIX_BYTES_string
#define IPFIX_LENGTH_mibModuleName                            IPFIX_BYTES_string
#define IPFIX_LENGTH_mobileIMSI                               IPFIX_BYTES_string
#define IPFIX_LENGTH_mobileMSISDN                             IPFIX_BYTES_string
#define IPFIX_LENGTH_httpRequestMethod                        IPFIX_BYTES_string
#define IPFIX_LENGTH_httpRequestHost                          IPFIX_BYTES_string
#define IPFIX_LENGTH_httpRequestTarget                        IPFIX_BYTES_string
#define IPFIX_LENGTH_httpMessageVersion                       IPFIX_BYTES_string
#define IPFIX_LENGTH_internalAddressRealm                     IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_externalAddressRealm                     IPFIX_BYTES_octetArray
#define IPFIX_LENGTH_httpUserAgent                            IPFIX_BYTES_string
#define IPFIX_LENGTH_httpContentType                          IPFIX_BYTES_string
#define IPFIX_LENGTH_httpReasonPhrase                         IPFIX_BYTES_string
```